### PR TITLE
feat(providers): Add `ExtendBlobRetention` support for Azure

### PR DESCRIPTION
**Pull request by KastenMike**
Adds `ExtendBlobRetention` code for Azure and fixes a couple typos I saw.

### CODE REVIEW

File: repo/blob/s3/s3_storage_test.go [hunk 1]
```diff
@@ -119,7 +119,7 @@ func getProviderOptions(tb testing.TB, envName string) *Options {
 	}
 
 	if o.Prefix != "" {
-		tb.Fatalf("options providd in '%v' must not specify a prefix", envName)
+		tb.Fatalf("options provided in '%v' must not specify a prefix", envName)
```
Comment (KastenMike) [inline repo/blob/s3/s3_storage_test.go:122]: Note: tb.Fatalf doesn't actually make the test fail. For that it should be followed by a tb.FailNow()

File: repo/blob/storage.go [hunk 2]
```diff
@@ -107,6 +108,9 @@ const (
 
 	// Compliance - compliance mode.
 	Compliance RetentionMode = "COMPLIANCE"
+
+	// Locked - Locked policy mode for Azure.
```
Comment (julio-lopez) [inline repo/blob/storage.go:112]: We should include _"unlocked"_ as well.

**Review by julio-lopez — APPROVED** [review]
This PR should go in with the one that depends on it as well, since that one has the tests :)

Merging this one now to move forward ;)

@KastenMike For future reference, it often makes sense to submit small trivial changes, such as typo corrections in their own PR. The reasons being - those are trivial to review and thus can be easily merged right away;
- and then they reduce noise in the PRs that have functional changes, so those become easier to review.

**Comment by julio-lopez** [comment]
@KastenMike also, thanks again for breaking things into smaller pieces to facilitate reviewing.

**Comment by julio-lopez** [comment]
@redgoat650 FYI

**Pull request merged**